packages/casadi/patches/0004-disable-deepbind.patch

```diff
@@ -0,0 +1,32 @@
+From 6c9443c0c1694600927dca29373d29fb741581f2 Mon Sep 17 00:00:00 2001
+From: Agriya Khetarpal <74401230+agriyakhetarpal@users.noreply.github.com>
+Date: Mon, 15 Jul 2024 18:40:51 +0530
+Subject: [PATCH 4/4] Disable RTLD_DEEPBIND by setting -DWITH_DEEPBIND to OFF
+
+RTLD_DEEPBIND gives symbol resolution preference to the loaded library
+over the global scope, it's usually used to avoid symbol collision. However,
+WebAssembly handles internal symbol resolution differently, so RTLD_DEEPBIND
+might not be necessary. This flag is used on Linux systems and does not
```

Comment:
> might not be necessary

Do we get a specific crash if we remove this patch? If you don't have a test case, let's drop it.

Reply:
It fails to compile for some reason, but I haven't looked too deeply into why. Let me try to compile without this patch in a moment and paste the logs here

Reply:
Never mind, I built with `WITH_DEEPBIND` set to `ON` but disabled `RTLD_DEEPBIND conditionally, and the build passed. I'll update the patch, I don't think it's supported outside of Linux and the Emscripten docs don't contain it either. Here's what I found about it: https://manpages.debian.org/dlopen(3)

Reply:
Resolving this conversation because I updated the patch, `RTLD_DEEPBIND` seems to be valid only for Linux...